flake.nix

```diff
@@ -0,0 +1,109 @@
+{
+  description = "A very basic flake";
+
+  nixConfig = {
+    flake-registry = "https://github.com/serokell/flake-registry/raw/master/flake-registry.json";
+  };
+
+  inputs = {
+    haskell-nix.url = "github:input-output-hk/haskell.nix";
+    nixpkgs.url = "github:serokell/nixpkgs";
+    hackage.flake = false;
+    stackage.flake = false;
+  };
+
+  outputs = { self, nixpkgs, flake-utils, haskell-nix, hackage, stackage, ... }@inputs:
+  flake-utils.lib.eachSystem ["x86_64-linux"] (system: let
+    pkgs = import inputs.nixpkgs {
+      inherit system;
+      overlays = [ inputs.haskell-nix.overlay ];
+    };
+
+    # run ci with resolver defined in stack.yaml and a newer resolver for servant 0.19
+    resolvers = [ null "lts-18.28" ];
```

Comment:
AFAICS, `lts-18.28` has the same `servant-client-core` version as currently used `lts-18.21`, see https://www.stackage.org/lts-18.28

Reply:
As an alternative, you can modify `stack.yaml` with `servant-client-core-0.19` extra dependency without modifying the resolver

Reply:
Would that play out well with all dependencies? I assumed servant-client-core would depend on newer versions of some packages that would not be available with the old resolver

Reply:
Adding both `servant-0.19` and `servant-client-core-0.19` to stack.yaml does seem to resolve correctly but leads to ci errors on the haskell side

> AFAICS, lts-18.28 has the same servant-client-core version as currently used lts-18.21, see https://www.stackage.org/lts-18.28

I had a typo in the flake, meant to put `19.28` :facepalm:  but that leads to another error saying that `text-format` cannot be resolved

Reply:
> As an alternative, you can modify stack.yaml with servant-client-core-0.19 extra dependency without modifying the resolver

I changed it over to that approach, seems more sensible. I have to admit I do not fully understand how dependencies are resolved outside the "main" resolver but it seems much more flexible than I had assumed